assets/js/modules/reader-revenue-manager/components/settings/SettingsEdit.test.js

```diff
@@ -96,4 +83,127 @@ describe( 'SettingsEdit', () => {
 		);
 	} );
+
+	it( 'should render the publication onboarding state notice if applicable', async () => {
+		const { getByText, waitForRegistry } = render( <SettingsEdit />, {
+			registry,
+		} );
+
+		await waitForRegistry();
+
+		expect(
+			getByText(
+				'Your publication requires further setup in Reader Revenue Manager'
+			)
+		).toBeInTheDocument();
+	} );
+
+	it( 'should not render the publication onboarding state notice if the user does not have module access', async () => {
+		registry.dispatch( MODULES_READER_REVENUE_MANAGER ).setOwnerID( 2 );
+
+		registry
+			.dispatch( CORE_MODULES )
+			.receiveCheckModuleAccess(
+				{ access: false },
+				{ slug: READER_REVENUE_MANAGER_MODULE_SLUG }
+			);
+
+		const { queryByText, waitForRegistry } = render( <SettingsEdit />, {
+			registry,
+		} );
+
+		await waitForRegistry();
+
+		expect(
+			queryByText(
+				'Your publication requires further setup in Reader Revenue Manager'
+			)
+		).not.toBeInTheDocument();
+	} );
+
+	it( 'should not render the publication onboarding state notice if the publication is not available', async () => {
+		registry
+			.dispatch( MODULES_READER_REVENUE_MANAGER )
+			.receiveGetPublications( [ publications[ 0 ], publications[ 1 ] ] );
+
+		const { queryByText, waitForRegistry } = render( <SettingsEdit />, {
+			registry,
+		} );
+
+		await waitForRegistry();
+
+		expect(
+			queryByText(
+				'Your publication requires further setup in Reader Revenue Manager'
+			)
+		).not.toBeInTheDocument();
+	} );
+
+	it( 'should render an error message if the publication is not available', async () => {
+		registry
+			.dispatch( MODULES_READER_REVENUE_MANAGER )
+			.receiveGetPublications( [ publications[ 0 ], publications[ 1 ] ] );
+
+		const { getByText, waitForRegistry } = render( <SettingsEdit />, {
+			registry,
+		} );
+
+		await waitForRegistry();
+
+		expect(
+			getByText(
+				`Error: The previously selected publication with ID ${ publicationID } was not found. Please select a new publication.`
+			)
+		).toBeInTheDocument();
+	} );
+
+	it( 'should not render an error message if the user does not have module access even if the publication is not available', async () => {
+		registry.dispatch( MODULES_READER_REVENUE_MANAGER ).setOwnerID( 2 );
+
+		registry
+			.dispatch( CORE_MODULES )
+			.receiveCheckModuleAccess(
+				{ access: false },
+				{ slug: READER_REVENUE_MANAGER_MODULE_SLUG }
+			);
+
+		registry
+			.dispatch( MODULES_READER_REVENUE_MANAGER )
+			.receiveGetPublications( [ publications[ 0 ], publications[ 1 ] ] );
+
+		const { queryByText, waitForRegistry } = render( <SettingsEdit />, {
+			registry,
+		} );
+
+		await waitForRegistry();
+
+		expect(
+			queryByText(
+				`Error: The previously selected publication with ID ${ publicationID } was not found. Please select a new publication.`
+			)
+		).not.toBeInTheDocument();
+	} );
+
+	it( 'should render an error notice if the user does not have module access', async () => {
```

Comment:
Technically, it's an error notice. However, visually, it's a warning. Since it's a unit test, let's keep it as it is.